traffic_ops/app/lib/UI/Topology.pm

```diff
@@ -264,7 +264,7 @@ sub gen_crconfig_json {
             $data_obj->{'contentServers'}->{ $row->host_name }->{'ip6'}           = ( $row->ip6_address || "" );
             $data_obj->{'contentServers'}->{ $row->host_name }->{'profile'}       = $row->profile->name;
             $data_obj->{'contentServers'}->{ $row->host_name }->{'type'}          = $row->type->name;
-            $data_obj->{'contentServers'}->{ $row->host_name }->{'hashId'}        = $row->xmpp_id;
+            $data_obj->{'contentServers'}->{ $row->host_name }->{'hashId'}        = $row->xmpp_id ? $row->xmpp_id : $row->host_name;
```

Comment:
would this allow an xmpp_id = "" which would probably not be a good thing. do we need a better test like isDefined and neq empty string?